src/firefly/js/tables/ui/BasicTableView.jsx

```diff
@@ -0,0 +1,233 @@
+        return (
+            <Resizable id='table-resizer' style={{position: 'relative', width: '100%', height: '100%', overflow: 'hidden'}} onResize={this.onResize}>
+                { textView ? <TextView { ...{columns, data, showUnits, heightPx, widthPx} }/> :
+                    <Table
+                        rowHeight={rowHeight}
+                        headerHeight={headerHeight}
+                        rowsCount={data.length}
+                        isColumnResizing={false}
+                        onColumnResizeEndCallback={this.onColumnResizeEndCallback}
+                        onRowClick={(e, index) => callbacks.onRowHighlight && callbacks.onRowHighlight(index)}
+                        rowClassNameGetter={this.rowClassName}
+                        scrollToRow={hlRowIdx}
+                        width={widthPx}
+                        height={heightPx}>
+                        { makeColumns(colProps) }
+                    </Table>
+                }
+                {showMask && <div style={{top: 0}} className='loading-mask'/>}
+                {isEmpty(data) && <div className='tablePanel_NoData'> No Data Found </div>}
+            </Resizable>
+        );
+    }
+}
+
+BasicTableView.propTypes = {
+    columns: PropTypes.arrayOf(PropTypes.object),
+    data: PropTypes.arrayOf(PropTypes.array),
+    hlRowIdx: PropTypes.number,
+    selectInfoCls: PropTypes.instanceOf(SelectInfo),
+    filterInfo: PropTypes.string,
+    sortInfo: PropTypes.string,
+    selectable: PropTypes.bool,
+    showUnits: PropTypes.bool,
+    showFilters: PropTypes.bool,
+    textView: PropTypes.bool,
+    rowHeight: PropTypes.number,
+    renderers: PropTypes.objectOf(
+        PropTypes.shape({
+            cellRenderer: PropTypes.func,
+            headRenderer: PropTypes.func
+        })
+    ),
+    callbacks: PropTypes.shape({
+        onRowHighlight: PropTypes.func,
+        onRowSelect: PropTypes.func,
+        onSelectAll: PropTypes.func,
+        onSort: PropTypes.func,
+        onFilter: PropTypes.func
+    })
+};
+
+BasicTableView.defaultProps = {
+    selectable: false,
+    showUnits: false,
+    showFilters: false,
+    rowHeight: 20
+};
+
+const TextView = ({columns, data, showUnits, widthPx, heightPx}) => {
+    const text = tableToText(columns, data, showUnits);
+    return (
+        <div style={{height: heightPx, width: widthPx,overflow: 'hidden'}}>
+            <div style={{height: '100%',overflow: 'auto'}}>
+                <pre>{text}</pre>
+            </div>
+        </div>
+    );
+};
+
+function makeColWidth(columns, data, showUnits) {
+    return !columns ? {} : columns.reduce((widths, col, cidx) => {
+        const label = col.name;
+        var nchar = col.prefWidth;
+        const unitLength = showUnits ? get(col, 'units.length', 0) : 0;
+        if (!nchar) {
+            nchar = Math.max(label.length, unitLength, get(data, `0.${cidx}.length`, 0));
+        }
```

Comment:
that'll work, too.  this is older code when i am trying to see which way is easier to read.. i.e.  'a.b.c.0'  or ['a', 'b', 'c', 0].